lib/rx/middleware.rb

```diff
@@ -34,12 +35,16 @@ def call(env)
       when "/readiness"
         readiness_response(check_to_component(readiness_checks))
       when "/deep"
-        @cache.cache("deep") do
-          readiness = check_to_component(readiness_checks)
-          critical  = check_to_component(deep_critical_checks)
-          secondary = check_to_component(deep_secondary_checks)
-
-          deep_response(readiness, critical, secondary)
+        if @options[:authorization] && !Rx::Util::HealthCheckAuthorization.new(env, @options[:authorization]).ok?
```

Comment:
what do you think about simplifying this check by allowing `HealthCheckAuthorization` to handle a `nil` authorization (by returning true)?

Reply:
True, it is applied. I think it is possible to use unless-else here since we are checking for falseness but I prefer going the if-else due to its readability. what do you think about it?